CIP-????/README.md

```diff
@@ -555,89 +571,101 @@ They thus represent real-world workloads.
+| multisig-sm-1            | 394.5 μs | 405.5 μs           | +2.8%  |
+| multisig-sm-2            | 385.2 μs | 392.8 μs           | +2.0%  |
+| multisig-sm-3            | 386.5 μs | 394.5 μs           | +2.1%  |
+| multisig-sm-4            | 385.8 μs | 404.2 μs           | +4.8%  |
+| multisig-sm-5            | 567.7 μs | 583.0 μs           | +2.7%  |
+| multisig-sm-6            | 391.7 μs | 405.5 μs           | +3.5%  |
+| multisig-sm-7            | 382.7 μs | 394.3 μs           | +3.0%  |
+| multisig-sm-8            | 389.3 μs | 395.8 μs           | +1.7%  |
+| multisig-sm-9            | 387.1 μs | 404.4 μs           | +4.5%  |
+| multisig-sm-10           | 567.4 μs | 583.3 μs           | +2.8%  |
+| ping-pong-1              | 320.6 μs | 327.4 μs           | +2.1%  |
+| ping-pong-2              | 319.1 μs | 327.0 μs           | +2.5%  |
+| ping-pong_2-1            | 182.2 μs | 190.6 μs           | +4.6%  |
+| prism-1                  | 139.2 μs | 150.2 μs           | +7.9%  |
+| prism-2                  | 404.5 μs | 412.2 μs           | +1.9%  |
+| prism-3                  | 339.4 μs | 359.4 μs           | +5.9%  |
+| pubkey-1                 | 118.6 μs | 123.7 μs           | +4.3%  |
+| stablecoin_1-1           | 962.4 μs | 976.3 μs           | +1.4%  |
+| stablecoin_1-2           | 163.7 μs | 174.2 μs           | +6.4%  |
+| stablecoin_1-3           | 1.103 ms | 1.120 ms           | +1.5%  |
+| stablecoin_1-4           | 174.0 μs | 185.4 μs           | +6.6%  |
+| stablecoin_1-5           | 1.391 ms | 1.418 ms           | +1.9%  |
+| stablecoin_1-6           | 215.4 μs | 227.5 μs           | +5.6%  |
+| stablecoin_2-1           | 962.4 μs | 981.3 μs           | +2.0%  |
+| stablecoin_2-2           | 163.5 μs | 174.2 μs           | +6.5%  |
+| stablecoin_2-3           | 1.099 ms | 1.117 ms           | +1.6%  |
+| stablecoin_2-4           | 173.6 μs | 184.8 μs           | +6.5%  |
+| token-account-1          | 174.1 μs | 181.2 μs           | +4.1%  |
+| token-account-2          | 312.8 μs | 334.3 μs           | +6.9%  |
+| uniswap-1                | 408.0 μs | 425.4 μs           | +4.3%  |
+| uniswap-2                | 203.9 μs | 211.7 μs           | +3.8%  |
+| uniswap-3                | 1.779 ms | 1.830 ms           | +2.9%  |
+| uniswap-4                | 282.6 μs | 297.6 μs           | +5.3%  |
+| uniswap-5                | 1.139 ms | 1.180 ms           | +3.6%  |
+| uniswap-6                | 276.2 μs | 288.4 μs           | +4.4%  |
+| vesting-1                | 339.8 μs | 356.0 μs           | +4.8%  |
+
+This is an average slowdown of 4%, which is not good at all. We do not want to have a negative impact on scripts that don't use the new constructs.
+
+However, this slowdown is very difficult to avoid.
+The GHC Core (GHC's intermediate language for Haskell programs) for both versions looks nearly identical, with the only differences being the introduction of new code for the new cases.
```

Comment:
File GHC bug? IIRC the magic number 8 or so, no? Intuitively it seems like you can raise it.

Reply:
It's 7 on 64 bit platforms. I'm looking further, but we're down to looking at the Cmm now...